source/common/http/conn_manager_impl.h

```diff
@@ -111,34 +112,26 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
 private:
   struct ActiveStream;
 
-  // Used to abstract making of RouteConfig update request.
-  // RdsRouteConfigUpdateRequester is used when an RdsRouteConfigProvider is configured,
-  // NullRouteConfigUpdateRequester is used in all other cases (specifically when
-  // ScopedRdsConfigProvider/InlineScopedRoutesConfigProvider is configured)
-  class RouteConfigUpdateRequester {
-  public:
-    virtual ~RouteConfigUpdateRequester() = default;
-    virtual void requestRouteConfigUpdate(const std::string, Event::Dispatcher&,
-                                          Http::RouteConfigUpdatedCallbackSharedPtr) {
-      NOT_IMPLEMENTED_GCOVR_EXCL_LINE;
-    };
-  };
-
-  class RdsRouteConfigUpdateRequester : public RouteConfigUpdateRequester {
+  class RdsRouteConfigUpdateRequester {
   public:
     RdsRouteConfigUpdateRequester(Router::RouteConfigProvider* route_config_provider)
         : route_config_provider_(route_config_provider) {}
-    void requestRouteConfigUpdate(
-        const std::string host_header, Event::Dispatcher& thread_local_dispatcher,
-        Http::RouteConfigUpdatedCallbackSharedPtr route_config_updated_cb) override;
+
+    // Only on demand update for ScopeRdsConfigProvider is supported.
+    // InlineScopedRoutesConfigProvider will cast to nullptr in this ctor.
+    RdsRouteConfigUpdateRequester(Config::ConfigProvider* scoped_route_config_provider)
+        : scoped_route_config_provider_(
+              dynamic_cast<Router::ScopedRdsConfigProvider*>(scoped_route_config_provider)) {}
+
+    void requestVhdsUpdate(const std::string host_header,
```

Comment:
Can we use const& of the host header here? I don see any need to copy the string. @dmitri-d

Reply:
sounds good to me.